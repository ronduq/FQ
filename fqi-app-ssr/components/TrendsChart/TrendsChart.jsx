import React from 'react';
import PropTypes from 'prop-types';
import { CartesianGrid, Line, LineChart, ReferenceLine, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts';
import moment from 'moment';

import CustomizedTimeline from '../CustomizedTimeline/CustomizedTimeline'

import s from './TrendsChart.scss';

const TrendsChart = ({data, handleSetDate, items, selectedDate, viewport, viewType, compareBy}) => {

  const formatXAxis = time => {
    const format = viewType === 'weekly' ? 'MMM D' : 'MMM';
    return moment(time).format(format);
  }

  const isViewportLarge = ['large', 'xlarge'].includes(viewport);
  const mobileOffsetLeft = !isViewportLarge ? -30 : 0;

  return (
    <div className={s.chart} aria-label="Produce trends chart">
      <ResponsiveContainer width="100%" height="100%">
        <LineChart data={data} margin={{ top: 0, right: 15, left: mobileOffsetLeft, bottom: 0 }} onClick={handleSetDate}>  
          <CartesianGrid strokeDasharray="2 6" fill="#F2F9F8"/>

          {items.map((item, i)  => {
            const { id, isParent, selected, colour = '#02917F' } = item;
            if (isParent || !selected) return;
            return <Line 
              isAnimationActive={true} 
              key={`line-${id}-${i}`} 
              activeDot={{ stroke: colour, fill: 'white', strokeWidth: 2, r: 5 }} 
              dot={{ stroke: colour, fill: colour, strokeWidth: 2, r: 5 }} 
              strokeWidth={2} 
              stroke={colour} 
              type="linear" 
              dataKey={id} 
            />
          })}
          
          <Tooltip content={<></>} />

          <XAxis dataKey="timestamp" width={10} tickFormatter={formatXAxis} domain={['auto', 'auto']}/>
          <YAxis domain={['auto', 'auto']} reversed={compareBy === 'quality' ? false : true }/>

          <ReferenceLine isFront={true} x={selectedDate}  label={<CustomizedTimeline />} />
        </LineChart>
      </ResponsiveContainer>
    </div>
  );
}

TrendsChart.propTypes = {
  data: PropTypes.array,
  handleSetDate: PropTypes.func,
  items: PropTypes.array,
  selectedDate: PropTypes.number,
  viewport: PropTypes.string,
  viewType: PropTypes.string,
};

TrendsChart.defaultProps = {
  viewType: 'weekly'
};

export default TrendsChart;




