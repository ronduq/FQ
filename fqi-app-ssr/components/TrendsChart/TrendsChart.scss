@import '../../style/settings/colours';
@import '../../style/settings/fonts';
@import '../../style/settings/breakpoints';

.chart {
  width: 100%;
  height: 250px;
  max-height: 250px;
  position: relative;

  @include respond-to('large', min-width) {
    height: 600px;
    max-height: 600px;
  }
}

:global {
  .recharts-cartesian-grid {
    cursor: pointer;
  }
  .recharts-cartesian-axis-tick text {
    @include font-size('xsmall');
  }
}

