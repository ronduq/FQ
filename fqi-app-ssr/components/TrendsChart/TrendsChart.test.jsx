/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import TrendsChart from './TrendsChart';

function createNodeMock() {
  const doc = document.implementation.createHTMLDocument();
  return { parentElement: doc.body };
}

describe('TrendsChart', () => {
  const props = {
    viewType: 'weeks',
    items:[{id: 'apple', isParent: true, selected: true}, {id: 'avocado', isParent: false, selected: true}, {id: 'tomato', isParent: false, selected: true}]
  }

  it('TrendsChart snapshot"', () => {
    const component = renderer.create(<TrendsChart {...props} />, { createNodeMock } );
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });

})