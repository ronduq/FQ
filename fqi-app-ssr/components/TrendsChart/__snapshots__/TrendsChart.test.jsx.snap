// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendsChart TrendsChart snapshot" 1`] = `
<div
  aria-label="Produce trends chart"
  className="chart"
>
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "none",
          "height": 0,
          "position": "absolute",
          "visibility": "hidden",
          "width": 0,
        }
      }
    />
  </div>
</div>
`;
